### PR TITLE
SN-516 improve handling of hidden properties

### PR DESCRIPTION
**Pull request by toanphamsaritasa**
https://saritasa.atlassian.net/browse/SN-516

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -432,3 +432,24 @@ You can configure your query for specific entity.
 ```
 
 You can use `ServiceProvider` to access your services.
+
+## Exclude property from query
+
+Users can explicitly control whether a property should be excluded from the data query.
```
Comment (AlexanderMakarov124) [inline README.md:438]: Let's continue writing documentation with the previous style
```suggestion
You can explicitly control whether a property should be excluded from the data query.
```

**Review by AlexanderMakarov124 — APPROVED** [review]

File: src/Saritasa.NetForge.Domain/Attributes/NetForgePropertyAttributeBase.cs [hunk 2]
```diff
@@ -10,6 +10,9 @@ public abstract class NetForgePropertyAttributeBase : Attribute
     /// <inheritdoc cref="PropertyMetadataBase.IsHidden"/>
     public bool IsHidden { get; set; }
 
+    /// <inheritdoc cref="PropertyMetadataBase.IsExcludeFromQuery"/>
+    public bool IsExcludeFromQuery { get; set; }
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Domain/Attributes/NetForgePropertyAttributeBase.cs:14]: `IsExcludedFromQuery`

File: src/Saritasa.NetForge.Domain/Entities/Metadata/PropertyMetadataBase.cs [hunk 3]
```diff
@@ -38,6 +38,11 @@ public abstract class PropertyMetadataBase
     /// </summary>
     public bool IsHidden { get; set; }
 
+    /// <summary>
+    /// Whether the property is exclude from the query.
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Domain/Entities/Metadata/PropertyMetadataBase.cs:42]:
```suggestion
    /// Whether the property is excluded from the query.
```

File: src/Saritasa.NetForge.Domain/Entities/Metadata/PropertyMetadataBase.cs [hunk 4]
```diff
@@ -38,6 +38,11 @@ public abstract class PropertyMetadataBase
...
+    /// </summary>
+    public bool IsExcludeFromQuery { get; set; }
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Domain/Entities/Metadata/PropertyMetadataBase.cs:44]: The same as above

File: src/Saritasa.NetForge.Domain/Entities/Options/PropertyOptions.cs [hunk 5]
```diff
@@ -14,6 +14,9 @@ public class PropertyOptions
     /// <inheritdoc cref="PropertyMetadataBase.IsHidden"/>
     public bool IsHidden { get; set; }
 
+    /// <inheritdoc cref="PropertyMetadataBase.IsExcludeFromQuery"/>
+    public bool IsExcludeFromQuery { get; set; }
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Domain/Entities/Options/PropertyOptions.cs:18]: The same as above

File: src/Saritasa.NetForge.DomainServices/PropertyOptionsBuilder.cs [hunk 6]
```diff
@@ -31,6 +31,15 @@ public PropertyOptionsBuilder SetIsHidden(bool isHidden)
         return this;
     }
 
+    /// <summary>
+    /// Sets whether the property should be exclude from the query.
+    /// </summary>
+    public PropertyOptionsBuilder SetIsExcludeFromQuery(bool isExcludeFromQuery)
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.DomainServices/PropertyOptionsBuilder.cs:37]:
```suggestion
    public PropertyOptionsBuilder SetIsExcludedFromQuery(bool isExcludeFromQuery)
```

File: src/Saritasa.NetForge.DomainServices/PropertyOptionsBuilder.cs [hunk 7]
```diff
@@ -31,6 +31,15 @@ public PropertyOptionsBuilder SetIsHidden(bool isHidden)
         return this;
     }
 
+    /// <summary>
+    /// Sets whether the property should be exclude from the query.
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.DomainServices/PropertyOptionsBuilder.cs:35]:
```suggestion
    /// Sets whether the property should be excluded from the query.
```

**Review by vasiliy-chefonov — APPROVED** [review]

**Pull request merged**